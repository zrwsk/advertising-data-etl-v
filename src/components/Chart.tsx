import { Box } from "@material-ui/core";
import { LineChart, CartesianGrid, XAxis, YAxis, Legend, Line, ResponsiveContainer } from "recharts";

const Chart = (): JSX.Element => {
  return (
    <Box mt={4} pr={8}>
      <ResponsiveContainer width="100%" height={400}>
        <LineChart data={[]}>
          <CartesianGrid vertical={false} />
          <XAxis />
          <YAxis />
          <Legend />
          <Line type="monotone" dataKey="Clicks" stroke="#8884d8" />
          <Line type="monotone" dataKey="Impressions" stroke="#82ca9d" />
        </LineChart>
      </ResponsiveContainer>
    </Box>
  );
};

export default Chart;