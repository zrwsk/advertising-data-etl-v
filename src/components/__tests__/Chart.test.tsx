import { render, screen } from '@testing-library/react';
import Chart from '../Chart';

jest.mock('../../utils/dataTransformation', () => ({
  getAggregatedData: () => [{
    clicks: 543592,
    date: '01.01.2020',
    impressions: 2574,
  },
  {
    clicks: 1000,
    date: '02.01.2020',
    impressions: 2112,
  },
  {
    clicks: 123,
    date: '01.02.2020',
    impressions: 3422,
  },
  {
    clicks: 4342,
    date: '03.03.2020',
    impressions: 5664,
  },
  {
    clicks: 5435435,
    date: '04.05.2020',
    impressions: 24324,
  }],
}));

describe('Chart component', () => {
  const filtersMock = {
    datasources: [],
    campaigns: [],
  };

  it('should render with main elements correctly', () => {
    const { asFragment } = render(<Chart data={[]} filters={filtersMock} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display active filters', () => {
    render(<Chart data={[]} filters={{ ...filtersMock, campaigns: ['C1'] }} />);
    expect(screen.getByText('All Datasources; Campaign "C1"')).toBeInTheDocument();
  });
});
