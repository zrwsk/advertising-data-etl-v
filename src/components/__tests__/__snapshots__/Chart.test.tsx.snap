// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart component should render with main elements correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <h2
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
    >
      All Datasources; All Campaigns
    </h2>
    <ul
      data-testid="chart-data"
    >
      <li>
        01.01.2020, 543592, 2574
      </li>
      <li>
        02.01.2020, 1000, 2112
      </li>
      <li>
        01.02.2020, 123, 3422
      </li>
      <li>
        03.03.2020, 4342, 5664
      </li>
      <li>
        04.05.2020, 5435435, 24324
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
